src/main/java/com/blackducksoftware/integration/hub/alert/config/CommonConfig.java

```diff
@@ -48,15 +57,39 @@
     protected final TaskExecutor taskExecutor;
     protected final NotificationRepository notificationRepository;
     protected final PlatformTransactionManager transactionManager;
+    private final TaskScheduler taskScheduler;
+
+    private ScheduledFuture<?> future;
 
     public CommonConfig(final SimpleJobLauncher jobLauncher, final JobBuilderFactory jobBuilderFactory, final StepBuilderFactory stepBuilderFactory, final TaskExecutor taskExecutor, final NotificationRepository notificationRepository,
-            final PlatformTransactionManager transactionManager) {
+            final PlatformTransactionManager transactionManager, final TaskScheduler taskScheduler) {
         this.jobLauncher = jobLauncher;
         this.jobBuilderFactory = jobBuilderFactory;
         this.stepBuilderFactory = stepBuilderFactory;
         this.taskExecutor = taskExecutor;
         this.notificationRepository = notificationRepository;
         this.transactionManager = transactionManager;
+        this.taskScheduler = taskScheduler;
+    }
+
+    public void scheduleJobExecution(final String cron) {
+        if (StringUtils.isNotBlank(cron)) {
```

Comment:
Should we have more validation on the cron expression? Even if we just split on spaces and counted the fields, that would make me feel a little better about this.

Reply:
I agree we should be validating them, but would you be ok with that validation being in the controllers? So if the User tries to configure gibberish we can send an error back .

Reply:
The post and put into the rest controller should validate the input values and return a correct error code.  I agree with @jrichardBD